### PR TITLE
Struct scalar from host dictionary

### PR DESCRIPTION
**Pull request by shaneding**
Allows the creation of a `struct_scalar` from python dictionary. Partly addresses #8558.

### CODE REVIEW

File: python/cudf/cudf/_lib/scalar.pyx [hunk 1]
```diff
@@ -308,6 +309,37 @@ cdef _set_decimal64_from_scalar(unique_ptr[scalar]& s,
...
+    if valid:
+        pyarrow_table = pa.Table.from_arrays(
+            [
+                pa.array([value[f.name]], from_pandas=True, type=f.type)
+                for f in arrow_schema
+            ],
+            names=columns
+        )
+    else:
+        # print("hello world")
```
Comment (brandon-b-miller) [inline python/cudf/cudf/_lib/scalar.pyx:327]:
```suggestion
```

File: python/cudf/cudf/_lib/scalar.pyx [hunk 2]
```diff
@@ -308,6 +309,36 @@ cdef _set_decimal64_from_scalar(unique_ptr[scalar]& s,
...
+                for f in arrow_schema
+            ],
+            names=columns
+        )
+
+    cdef Table table = from_arrow(pyarrow_table, column_names=columns)
+    cdef table_view struct_view = table.view()
+
+    s.reset(
+        new struct_scalar(struct_view)
```
Comment (brandon-b-miller) [inline python/cudf/cudf/_lib/scalar.pyx:339]: Missing `valid` here?

File: python/cudf/cudf/tests/test_struct.py [hunk 3]
```diff
@@ -98,3 +98,18 @@ def test_serialize_struct_dtype(fields):
...
     assert sr[0] == expected
+
+
+@pytest.mark.parametrize(
+    "data",
+    [
+        {"a": 1, "b": "rapids", "c": [1, 2, 3, 4]},
+        {"a": 1, "b": "rapids", "c": [1, 2, 3, 4], "d": cudf.NA},
+        {"a": "Hello"},
+        {"b": [], "c": [1, 2, 3]},
```
Comment (brandon-b-miller) [inline python/cudf/cudf/tests/test_struct.py:109]: We also want to make sure we can construct an `NA` scalar here so long as we pass a valid `StructDtype`.

**Review by brandon-b-miller — CHANGES_REQUESTED** [review]
Couple of questions but looking great.

**Review by brandon-b-miller — APPROVED** [review]

**Review by rgsl888prabhu — APPROVED** [review]

**Comment by shaneding** [comment]
@gpucibot merge

**Comment by brandon-b-miller** [comment]
@gpucibot merge

**Pull request merged**